hypershift-operator/main.go

```diff
@@ -367,7 +368,12 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
 	}).SetupWithManager(mgr); err != nil {
 		return fmt.Errorf("unable to create controller: %w", err)
 	}
-	npmetrics.CreateAndRegisterNodePoolsMetricsCollector(mgr.GetClient())
+	{
+		awsSession := awsutil.NewSession("hypershift-operator", "", "", "", "")
+		awsConfig := awsutil.NewConfig()
+
+		npmetrics.CreateAndRegisterNodePoolsMetricsCollector(mgr.GetClient(), ec2.New(awsSession, awsConfig))
```

Comment:
This should only be creating an aws client if the HO is running on AWS platform

Reply:
can you confirm the HO perms in ocm are suitable for this requests?

Reply:
can you point the source of truth defining HO aws perms in OCM?

Reply:
Yes, here are the HO AWS perms:
https://gitlab.cee.redhat.com/service/osd-fleet-manager/-/blob/main/config/roles/hypershift-operator-policy.json
I will need to add the `ec2:DescribeInstanceTypes` there.

Reply:
Here the MR adding that action:
https://gitlab.cee.redhat.com/service/osd-fleet-manager/-/merge_requests/723